src/libs/NetworkConnection.ts

```diff
@@ -118,20 +118,20 @@ function subscribeToBackendAndInternetReachability(): () => void {
             })
             .then((isBackendReachable: boolean) => {
                 if (isBackendReachable) {
-                    NetworkActions.setIsBackendReachable(true);
+                    NetworkActions.setIsBackendReachable(true, 'successfully completed API request');
                     return;
                 }
                 checkInternetReachability().then((isInternetReachable: boolean) => {
                     setOfflineStatus(!isInternetReachable);
                     setNetWorkStatus(isInternetReachable);
-                    NetworkActions.setIsBackendReachable(false);
+                    NetworkActions.setIsBackendReachable(false, 'request succeeded, but internet reachability test failed');
```

Comment:
This doesn't have anything to do with the reachability test, so I think it should be moved outside the promise (same as below). I think the intention is to always set it to false if the ping fails for any reason, but that doesn't mean the internet is reachable or not.